File: H/submissions/accepted/H.kt
fun main() {
    val (n, m, _) = readLine()!!.split(" ").map{ it.toInt() }
    val training = HashSet<List<Int>>()
    for (i in 1..n) {
        training.add(readLine()!!.split(" ").map{ it.toInt()})
    }
    for (i in 1..m) {
        val buf = readLine()!!.split(" ").map{ it.toInt() }
        if (training.contains(buf))
            println("BAD")
        else
            println("GOOD")
    }
}

File: H/submissions/accepted/H.cpp
#include<bits/stdc++.h>

using namespace std;

int main() {
    static char buf[104857600] = {};
    fgets(buf, 104857600, stdin);
    int n, m, d;
    sscanf(buf, "%d %d %d",&n, &m, &d);
    unordered_set<string> training;
    for (int i = 0; i < n; i++) {
        fgets(buf, 104857600, stdin);
        training.insert(string(buf));
    }
    for (int i = 0; i < m; i++) {
        fgets(buf, 104857600, stdin);
        if (training.count(string(buf))) puts("BAD");
        else puts("GOOD");
    }
    return 0;
}
